src/orderbook_fetcher.ts

```diff
@@ -29,6 +29,22 @@ export class OrderbookFetcher {
     poll()
   }
 
+  serializeOrderbooks(): object {
+    const o: any = {}
+    this.orderbooks.forEach((value, key) => {
+      o[key] = value.toJSON()
+    })
+    return o
+  }
+
+  static deserializeOrderbooks(o: any): Map<string, Orderbook> {
+    const orderbooks = new Map()
+    for (const [key, value] of Object.entries(o)) {
+      orderbooks.set(key, Orderbook.fromJSON(value))
+    }
+    return orderbooks
```

Comment:
This is not working on my machine. Fractions are serialized to something like:

`{ numerator: { negative: 0, words: [Array], length: 1, red: null }, denominator: { negative: 0, words: [Array], length: 1, red: null } }` which then cannot be deserialized using `new BN(o.denominator, "hex");`. I need to call `JSON.stringify` & `JSON.parse` to make it work.

Do you have an idea why it is working for you?

Reply:
Maybe different node JS versions?

Reply:
So it works on travis and for me but not for you. That IS weird. My yarn version is `1.21.1` and my node version is `10.18.1`.

Reply:
Maybe `BN` has released a new semver compatible version (even though it really isn't) that changes how BNs are serialized. We use the default serialization and expect it to serialize a BN to a hex string (see Fraction::fromJSON). On your machine the different BN dependency is used than on my machine and travis.

Reply:
BN seems fine. Maybe we do have different node versions and on mine it has to stringify things when passing to the workerpool but not on yours. We can probably fix this by always doing stringify and parse.

Reply:
I'm running yarn 1.17.3 and node v11.15.0

Reply:
I pushed a new change that I hope fixes it on your node version too. Also changed the test cleanup slightly which silenced a jest warning I got on my machine.

Reply:
@fleupold let me know if it works for you now